### DISCUSSION TITLE
Obsidian compatibility and Markor

### DISCUSSION THREAD

**Discussion by harshad1** [gsantner/markor#2137] [General] [answered]
[Obsidian](https://obsidian.md/) is a really popular note taking app which works on top of markdown files. It follows the same general philosophy as markor - work on plain text files in a native directory tree.

As many users use Markor together with Obsidian, it would be nice to try to get them to play well together where possible and resonable.

I am opening this discussion as a place to pull together

1. Ideas for features in Markor (I will implement the ones which make the most sense for me)
2. Hacks / Tips / Tricks

Just to make this clear I do not want to turn Markor into some sort of Obsidian client (and I am pretty sure @gsantner wouldn't agree anyway). For example, I see no need to support obsidian style links alongside regular markdown style links.

**Comment by harshad1** [comment]
How to imitate Obsidian's daily note feature: 

Markor will execute any JS code in a script tags when you switch to view mode

I use the following code to open a file with today's date in the current folder. If the file does not exist it will be created.

You can modify this to open a new file in the current folder with any programmatically generated name. If you want a new unique name you could use seconds or a UUID

```html
<script>
let anchor = document.createElement('a');

// This asinine approach is from https://stackoverflow.com/a/28149561
// offset in milliseconds
let offset = (new Date()).getTimezoneOffset() * 60000;

let iso = (new Date(Date.now() - offset)).toISOString()
    
let today = iso.split('T')[0]

let textNode = document.createTextNode("Today's worklog");

// Append the textNode as a child to anchor.
anchor.appendChild(textNode);

anchor.href = "./" + today + ".md";

anchor.style.fontSize = "x-large";

document.body.appendChild(anchor);

window.location.href = anchor.href

</script>
```

**Comment by gsantner** [comment, marked as answer]
Like mentioned, yes I don't want to turn Markor in Obsidian software. While Obsidian also happens to use Markdown, it is commercial software and has a lot of vendor specific (Obsidian specific) functionalities that are not available with most other Markdown software/parsers/clients. Obsidian has their own non-open-source client software, maintains it on their own and can pay employees to do so.

BUT yes of course collecting some relevant information here is a good idea 😄 .

**Comment by mendhak** [comment]
Obsidian is quite plugin and sync friendly but I can imagine that would be a huge burden for Markor maintenance. I wonder if Markor could support some kind of an **intent broadcast**, so that we could get other software (like Tasker, Llama Automate) to listen and do 'something' based on that event.  

Example, we could listen for the "file saved" event and kick off a script to sync the files to github, gdrive, etc. 

This would remove the burden of becoming a sync tool or plugin tool on Markor, and the broadcast just serves as a handoff. I've been doing something similar with [my own app](https://gpslogger.app/#gpsloggereventsbroadcast).

**Reply by harshad1** [reply]
Since markor works on local files I have had a lot of success using termux to run python scripts which do things.

I can see how some sort of intent system being useful. IMO designing the 'API' is the hard part, implementation will be easy enough.

If you are willing to put together some ideas, that would be a good step

**Comment by ekovarski** [comment]
As I came across this thread when I searched for a Markdown editor to use with Obsidian, I figured I'd add the following script to be used with Quick Note in case someone wants to create notes that follow the default core Unique Note Creator plugin naming format. It should also be relatively easy to adjust it for any other format you might want. Tested on Markor v2.14.1

```
<script>
    const date = new Date();

    // month is zero indexed, + 1
    // zero pad values
    const [month, day, year] = [ 
        ('0' + (date.getMonth() + 1)).slice(-2), 
        ('0' + date.getDate()).slice(-2),
        date.getFullYear(),
    ];
    const [hour, minutes, seconds] = [ 
        ('0' + date.getHours()).slice(-2), 
        ('0' + date.getMinutes()).slice(-2), 
        ('0' + date.getSeconds()).slice(-2),
    ];

    // use the default unique note creator format YYYY-MM-DD HHmm 
    const formattedTime = year + "-" + month + "-" + day + " " + hour + minutes;

    const path = "./" + formattedTime + ".md";

    window.location.href = path;
</script>
```

**Comment by leoncrf** [comment]
It would be great to be able to view and even edit MD files in the widget. The Todo widget only displays txt files, not MD files. Furthermore, the ToDo widget tends to add lines, which should be optional.

The second widget that lets you view files in folders could also has a option to let you access and edit the files in the widget itself instead of opening the app. 

Furthermore, I congratulate the team for their work and effort.
